cmd/proxy/Dockerfile

```diff
@@ -1,15 +1,34 @@
-FROM golang:1.11
+# Links to compare against to ensure we have all VCS's setup in this build
+# https://github.com/docker-library/buildpack-deps/blob/1845b3f918f69b4c97912b0d4d68a5658458e84f/stretch/scm/Dockerfile
+# https://github.com/golang/go/blob/f082dbfd4f23b0c95ee1de5c2b091dad2ff6d930/src/cmd/go/internal/get/vcs.go#L90
+
+FROM golang:1.11-alpine AS builder
 
-RUN mkdir -p $GOPATH/src/github.com/gomods/athens
 WORKDIR $GOPATH/src/github.com/gomods/athens
 
 COPY . .
 
-RUN CGO_ENABLED=0 GO111MODULE=on go build -mod=vendor -o /bin/app ./cmd/proxy
-RUN ./scripts/create_default_config.sh
+RUN apk update && \ 
+	apk add --no-cache bash && \
+	GO111MODULE=on CGO_ENABLED=0 go build -mod=vendor -o /bin/app ./cmd/proxy && \
+	./scripts/create_default_config.sh
+
+FROM alpine
+
+ENV GO111MODULE=on
+
+COPY --from=builder /bin/app /bin/app
+COPY --from=builder /go/src/github.com/gomods/athens/config.toml /config/config.toml
+COPY --from=builder /go/src/github.com/gomods/athens/cmd/proxy/locales /go/src/github.com/gomods/athens/cmd/proxy/locales
```

Comment:
Resource files should not be copied to the go source directory and should be copied to the working directory.
Athens uses the gobuffalo package. Normally, it should be published using `buffalo build`, which will package the resources.
If we don't use `buffalo build` then we need to copy the resources ourselves.

Here is the location of the settings:

https://github.com/gomods/athens/blob/c740fcb26cc0192391c10b4b0185d62b78c1410b/cmd/proxy/actions/app.go#L36
https://github.com/gomods/athens/blob/c740fcb26cc0192391c10b4b0185d62b78c1410b/cmd/proxy/actions/render.go#L9

This position is relative to PWD, which is the WORKDIR in the dockerfile.

Reply:
@tnextday couple notes: 
1. buffalo build is not playing well with modules right now, so `go build` is sufficient for the Proxy (I think buffalo just released a fix very recently) 
2. We don't really have any resources. In my opinion we should remove all this stuff in another pr.